docs/architecture/views.md

```diff
@@ -0,0 +1,169 @@
+# Views
+
+A **view** is a declarative description of what should get rendered and is usually influenced by the current [state](state.md).
+
+It is implemented as a pure function that accepts the current state and returns a [virtual DOM node (VNode)](#virtual-dom). When [state transitions](state.md#state-transitions) happen your views are automatically updated accordingly.
+
+---
+
+## Describing Views
+
+The [`h()`](../api/h.md), [`text()`](../api/text.md), and [`memo()`](../api/memo.md) functions are the building blocks of your views.
+
+[`h()`](../api/h.md) not only describes what HTML elements are being used but also what [actions](actions.md) are wired up if any.
+
+```js
+const view = (state) =>
+  h(
+    "button",
+    {
+      class: { "calling-acid-burn": state.beingFramed },
+      onclick: FindThatDisk,
+    },
+    text("It's in that place where I put that thing that time.")
+  )
+```
+<!-- In the 1995 movie "Hackers", the hacker "The Phantom Freak" calls his friend "Acid Burn" from jail as he's being framed for a crime he didn't commit. -->
+
+[`text()`](../api/text.md) just creates text nodes so the views it can create on its own are necessarily simplistic.
+
+```js
+const view = () => text("Go home and be a family man!")
+```
+<!-- In the videogame "Street Fighter II: The World Warrior", the fighter known as Guile says this taunt to his opponent after defeating them. -->
+
+[`memo()`](../api/memo.md) is designed to be used with other functions that produce VNodes, so it doesn't really describe a view by itself.
+
+```js
+const view = (state) => memo(scenicView, state.vacationSpot)
+```
+<!-- Just a play-on-words between how "view" is used in Hyperapp and everyday language. -->
+
+---
+
+## Components
+
+Views are naturally composable so they can be as simple or complicated as you need. Simpler apps probably just need a single view but in more complicated apps there could be plenty of subviews.
+
+A **component** in Hyperapp can either be a subview or some other function that generates VNodes.
```

Comment:
What I was going for at the time was to say that view functions always have a state parameter. So, a subview is just a view, including its **required** state parameter, that gets embedded within some other VNode-producing function, which might be another view or a component. Components are also embedded but are not limited to requiring a state parameter.

So, the original idea was that views and subviews were more app-specific while components were more reusable because if something needed to read the state to be able to do its thing then it needed to know about the app's state and therefore be app-specific.

That said, I'm open to reconsidering this.

As for "view component" I like it better than "component". It's less vague as you've pointed out. I'll use it.

Reply:
Let's go with view component. 🙆‍♂️

Reply:
@icylace you make a good point that there is a distinct difference between the sort of view-components that are reusable in any app, and the kind that are like "tailor made" for the given particular app (which are meant to distribute state properties to generic view-component properties or `h` properties). 

However, I feel like that difference is more obvious once your app gets big and you start thinking about how to divy up your view. In other words, it's a question of architecture and design patterns, and a little too abstract to bring up in reference documentation. 

I'm just worried that people will think that "sub-views" (app specific) and "view components" (generic reusable) are different concepts technically, in the view of hyperapp.

Generally speaking I am wary of bringing up too many concepts that relate to patterns rather than concrete features, as it can gives the impression that hyperapp is more complicated than it actually is.

There is nothing wrong with explaining concepts, such as the difference between a sub-view and a generic view component, but it should be in some separate more "how-to" oriented document. Kind of how React's doc don't explain higher-order-components in the reference docs for components, but they do bring it up in the article about "how do I share state between components"

Reply:
These sound like good reasons to rethink the view-related concepts, which I will do.

Reply:
I think it's fine to explain the concepts here until such article exists.